lib/network_utils.pm

```diff
@@ -51,16 +50,36 @@ Returns if can ping worker host gateway
 =cut
 sub can_upload_logs {
     my ($gw) = @_;
-    $gw ||= $default_gw;
-    return script_run('ping -c 1 ' . $default_gw);
+    $gw ||= testapi::host_ip();
+    return (script_run('ping -c 1 ' . $gw) == 0);
 }
 
 =head2 check_and_recover_network
 Recover network with static config if is feasible, returns if can ping GW
+Main use case is post_fail_hook, to be able to upload logs
 =cut
 sub recover_network {
-    setup_static_network(gw => testapi::host_ip());
-    return can_upload_logs;
+    my (%args) = @_;
```

Comment:
we don't need `skip_recovery` param. this function does not make sense when this param is true. So whenever we need to call it with `true` we should not call it :)

Reply:
Previously you've suggested to move that parameter to the method, which I did because without your support this PR will never get merged (that was my feeling). So I will change it back =)